File: 3 JSON Parser HTTP/C++ Parser+HTTP iOS/JSONNet-Poco/LocationType.cpp
//LocationType.cpp
#include "LocationType.h"


LocationType::LocationType(){
}
LocationType::~LocationType(){
}

LocationType::LocationType(Object::Ptr locationRaw){
    deserialize(locationRaw);
}

double LocationType::getLongitude() {
    return this->longitude;
}

void LocationType::setLongitude(double longitude) {
    this->longitude = longitude;
}

double LocationType::getLatitude() {
    return this->latitude;
}

void LocationType::setLatitude(double latitude) {
    this->latitude = latitude;
}

void LocationType::deserialize(Object::Ptr object){

    if(object->has(LONGITUDE) && !object->isNull(LONGITUDE)){
        double d;
        Var longitude = object->get(LONGITUDE);
        Object::Ptr subObject = longitude.extract<Object::Ptr>();
        if (subObject->has("$")&& !subObject->isNull("$")) {
            d = subObject->get("$").convert<double>();
        }else{
            d = longitude.convert<double>();
        }
        this->longitude = d;
    }
    if(object->has(LATITUDE) && !object->isNull(LATITUDE)){
        double d;
        Var LatRaw = object->get(LATITUDE);
        Var longitude = object->get(LATITUDE);
        Object::Ptr subObject = longitude.extract<Object::Ptr>();
        if (subObject->has("$")&& !subObject->isNull("$")) {
            d = subObject->get("$").convert<double>();
        }else{
            d = longitude.convert<double>();
        }
        this->latitude = d;
    }
}

File: 2 JSON Parser/Original Parser Android/src/com/example/javajsonparser/LocationType.java
//LocationType.java
package com.example.javajsonparser;

import java.io.IOException;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonToken;

public class LocationType {
    private static final String LATITUDE = "Latitude";
    private static final String LONGITUDE = "Longitude";

    private double longitude;
    private double latitude;

    public LocationType() {
    }

    public LocationType(final JsonReader reader) throws IOException {
        deserialize(reader);
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(final double longitude) {
        this.longitude = longitude;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(final double latitude) {
        this.latitude = latitude;
    }

    public void deserialize(final JsonReader reader) throws IOException {
        reader.beginObject();

        while (reader.hasNext()) {
            if (reader.peek() == JsonToken.NAME) {
                String name = reader.nextName();

                if (LATITUDE.equalsIgnoreCase(name)) {
                    reader.beginObject();
                    reader.nextName();
                    this.latitude = reader.nextDouble();
                    reader.endObject();
                }

                if (LONGITUDE.equalsIgnoreCase(name)) {
                    reader.beginObject();
                    reader.nextName();
                    this.longitude = reader.nextDouble();
                    reader.endObject();
                }
            } else {
                reader.skipValue();
            }
        }
        reader.endObject();
    }
}
